lcm/lcm.go

```diff
@@ -23,6 +23,59 @@ type Controller interface {
 	// More...
 }
 
-// TODO
 type CRStatus struct {
+	Phase      Phase       `json:"phase"`
+	Properties []*Property `json:"properties"`
+}
+
+//Phase is the current state of component
+type Phase string
+
+const (
+	//PendingPhase represents pending state of component.
+	//The reconcile has created CR, but the CR has not been scheduled yet.
+	PendingPhase Phase = "Pending"
+	//DeployingPhase represents deploying state of component.
+	//The reconcile is deploying CR, but the service is not available.
+	DeployingPhase Phase = "Deploying"
```

Comment:
What is the different between `Pending` and `Deploying`, When we call `Provision` method of Reconciler, which `Phase` will be returned.

Reply:
When the cache reconciler executes, it will initialize a status, this status is `Pending`. Next step reconciler will exec `Provision` method, the status at this time is `Deploying`.

Reply:
Maybe `Provision ` should include `Deploy` and `Check` two-step.